DEVELOPMENT.md

```diff
@@ -0,0 +1,97 @@
+## Run services with `go run`
+
+This section explains how to run some of the services using `go run` instead of the Docker setup to facilitate development.
+
+### Running `sda-download` with `go run`
+1. Bring up all SDA services with the S3 backend and populate them with test data by running the following command in the root folder of the repository:
+```sh
+make integrationtest-sda-s3-run 
+```
+
+2. Change to the folder `sda-download` and then create the credentials using: 
+```sh
+pushd dev_utils
+bash make_certs.sh
+chmod 600 certs/*-key.*
+popd
+```
+
+3. Update your `/etc/hosts` file: Add the following line to ensure the service can resolve OIDC locally: 
+```
+127.0.0.1 oidc
+```
+
+4. Start the `sda-download` service using:
+```sh
+APP_PORT=8553 CONFIGFILE=dev_utils/config-notls_local.yaml go run cmd/main.go
+```
+
+5. Check if `sda-download` works as expected using:
+```sh
+token=$(curl -s -k http://localhost:8080/tokens | jq -r '.[0]') 
+curl -H "Authorization: Bearer $token" http://localhost:8553/metadata/datasets
+```
+If successful, the curl command should output a JSON body containing:
+```json
+["EGAD74900000101"]
+```
+
+
+### Running other SDA services with `go run`
+Running other SDA services located in the `sda` subfolder, such as `ingest` or `verify`, differs slightly from running the `sda-download` service. Here, we'll use `ingest` as an example.
+
+1. Bring up all SDA services with the S3 backend and populate them with test data by running the following command in the root folder of the repository (this step is the same as for `sda-download`):
+```sh
+make integrationtest-sda-s3-run 
+```
+
+2. Bring down the `ingest` service using:
+```sh
+PR_NUMBER=$(/bin/date +%F)  docker compose -f .github/integration/sda-s3-integration.yml stop ingest  
+```
+
+3. Change to the folder `sda`, then copy keys and other information from the shared folder of the container using:
+```sh
+mkdir -p tmp
+docker cp verify:/shared tmp/
+```
```

Comment:
Do not copy temp files into the current workdir, always use the system temp folder.

```suggestion
3. Change to the folder `sda`, then copy keys and other information from the shared folder of the container using:
```sh
docker cp verify:/shared /tmp/shared
```
```